pkg/pillar/cmd/volumemgr/README.md

```diff
@@ -0,0 +1,149 @@
+Verifier alone has the right to modify files in directories under its control.
+
+Since each one controls its own location, they can change.
+
+In practice, to date, these have been the directories:
+
+* Downloads: `/persist/downloads/{appImg.obj,baseOs.obj}/pending`
+* Temporary cache for images undergoing verification: `/persist/downloads/{appImg.obj,baseOs.obj}/verifier`
+* Verified: `/persist/downloads/{appImg.obj,baseOs.obj}/verified`
+
+### Flow Inside Volume Manager
+
+The flow inside volumemgr is as follows. We separate the steps for downloading
+an app or downloading the baseos, but the flow is very similar.
+
+It is important to distinguish between when a request is made for a content tree
+that already exists and has been verified, and between a request for one that
+does not yet exist.
+
+#### Application Image
+
+On startup, volumemgr registers to receive notifications from agent `"zedmanager"`
+of scope `types.AppImgObj`, i.e. notifications for application image objects.
+It registers several handlers, the relevant one of which is `handleAppImgCreate`.
+
+Whenever zedmanager requests the download of an application image that does not
+yet exist, it creates the `appImgObj`. This triggers the handler `handleAppImgCreate`.
+
+1. `handleAppImgCreate()` is triggered. This is just a handler for the event.
+   1. `handleAppImgCreate()` calls `vcCreate()`
+1. `vcCreate()`:
+   1. creates a `VolumeStatus{}` and publishes it.
+   1. `vcCreate()` calls `doUpdate()`
+1. `doUpdate()` is called by `vcCreate()` as well as any time `updateVolumeStatus()` is called. `doUpdate()` is responsible for checking the status of a volume, based on its `VolumeStatus`, and then taking next steps, if needed. It is a "switchboard" for `VolumeStatus` updates.
+   * If the image is verified and the volume is created, we are done
+   * If the image is verified, create the volume
+   * If the image is downloaded, start verifying (see below)
+   * If the image is not yet downloaded, start the downloader (see below)
+
+##### Downloads
+
+The download is started by calling `AddOrRefcountDownloaderConfig()` which
+creates and publishes `types.DownloaderConfig`; the downloader is listening for
+this message to start a download.
+
+The downloader indicates its status, including completion, by publishing
+`types.DownloaderStatus`. Volume Manager registers the handler
+`handleDownloaderStatusModify` to catch these events.
+
+##### Verification
+
+The verification is started by calling `kickVerifier()` which calls
+`MaybeAddVerifyImageConfig()`, which creates and publishes
+`types.VerifyImageConfig`; The verifier is lisenting for this message to verify
+a file.
+
+The verifier indicates its status, including completion, by publishing
+`types.VerifyImageStatus`. Volume Manager registers the handler
+`handleVerifyImageStatusModify` to catch these events.
+
+##### Switchboard
+
+The "switchboard" is called whenever there is an update.
```

Comment:
Why introduce the new term "switchboard"? It appears nowhere in the code hence it might be a source of confusion for a newcommer.

Reply:
It was more of the concept. I can leave it as an idea, but remove it from the headers.